lib/browser/rpc-server.js

```diff
@@ -519,27 +520,45 @@ if (features.isDesktopCapturerEnabled()) {
 const getPreloadScript = async function (preloadPath) {
   let preloadSrc = null
   let preloadError = null
-  if (preloadPath) {
-    try {
-      preloadSrc = (await fs.promises.readFile(preloadPath)).toString()
-    } catch (err) {
-      preloadError = errorUtils.serialize(err)
-    }
+  try {
+    preloadSrc = (await fs.promises.readFile(preloadPath)).toString()
+  } catch (err) {
+    preloadError = errorUtils.serialize(err)
   }
   return { preloadPath, preloadSrc, preloadError }
 }
 
-ipcMainUtils.handle('ELECTRON_GET_CONTENT_SCRIPTS', () => getContentScripts())
+const getPreloadPaths = function (contents) {
+  const result = []
 
-ipcMainUtils.handle('ELECTRON_BROWSER_SANDBOX_LOAD', async function (event) {
-  const preloadPaths = [
-    ...(event.sender.session ? event.sender.session.getPreloads() : []),
-    event.sender._getPreloadPath()
-  ]
+  for (const preload of contents.session ? contents.session.getPreloads() : []) {
+    if (path.isAbsolute(preload)) {
+      result.push(preload)
+    } else {
+      console.error(`preload script must have absolute path: ${preload}`)
+    }
+  }
+
+  const preload = contents._getPreloadPath()
+  if (preload) {
+    result.push(preload)
+  }
 
+  return result
+}
+
+ipcMainUtils.handle('ELECTRON_INIT_RENDERER', async function (event) {
```

Comment:
The `async` is unnecessary here as there is no async operations, adding `async` would make a difference with https://github.com/electron/electron/pull/18990.

I do hope to getting #18990 merged before adding more usages of `ipcMainUtils.handle`.